### PR TITLE
Fixed warnings

### PR DESCRIPTION
**Pull request by tivervac**
Signed-off-by: Titouan Vervack <titouan.vervack@sigasi.com>

### CODE REVIEW

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend [hunk 1]
```diff
@@ -148,6 +148,7 @@ class ConstantConditionsInterpreter {
 								JvmIdentifiableElement,
 								ThisReference: <Object>newArrayList(v)
 								List<Object>: new ArrayList(v)
+								default: emptyList
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend]: am not sure if a immutable is a good idea here. @szarnekow what do you think?

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/DefaultReentrantTypeResolver.java [hunk 2]
```diff
@@ -142,7 +142,8 @@ public IResolvedTypes reentrantResolve(CancelIndicator monitor) {
 			if (operationCanceledManager.isOperationCanceledException(e)) {
 				operationCanceledManager.propagateAsErrorIfCancelException(e);
 			}
-			throw Throwables.propagate(e);
+			Throwables.throwIfUnchecked(e);
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/DefaultReentrantTypeResolver.java]: wont work with guava 14-19

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend [hunk 3]
```diff
@@ -202,15 +202,16 @@ class UnresolvableFeatureException extends ConstantExpressionEvaluationException
 	
 }
 
+@Data
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend]: is there a reason for adding this?

Comment (tivervac) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend]: The classes' fields were unused, adding an @Data adds accessors to the fields making them non-useless

Comment (cdietrich) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend]: 👍

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend]: @szarnekow ping

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend (hunk 3 above)
Comment (szarnekow) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend]: Hmm JvmField would not be serializable but exceptions are serializable. I'd rather remove the field and set the root cause to be the given cause.

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend (hunk 1 above)
Comment (tivervac) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend]: I'll fix this sometime next week

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend (hunk 3 above)
Comment (tivervac) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/AbstractConstantExpressionsInterpreter.xtend]: I'll fix this sometime next week

**Comment by szarnekow** [comment]
Did you find the time to work on this?

**Comment by tivervac** [comment]
@szarnekow I'll probably be able to fix it on monday.

**Pull request merged**